### DISCUSSION TITLE
Need help with plotting Taylor Factors in ODF sections

### DISCUSSION THREAD

**Discussion by HPouraliakbar** [mtex-toolbox/mtex#2182] [Ask Anything]
Hi MTEX community,

I need help plotting the Taylor factors calculated using a uniaxial strain tensor for all my indexed orientations within ODF-Phi2 sections. Can someone assist me with the code for this?

Thanks in advance for your help.
Hesam

**Comment by ralfHielscher** [comment]
Hi Hesam,

have a look here [Taylor model](https://mtex-toolbox.github.io/TaylorModel.html)

in short:

```matlab
cs = crystalSymmetry('432');
sS = slipSystem.fcc(cs)

eps = strainRateTensor.uniaxial

M = calcTaylor(eps, sS)

plot(M)
```

Ralf.

**Comment by HPouraliakbar** [comment]
Thank Ralf. 

This is just getting the M factor. I would like to plot the M factors for each orientation in ODF sections (lets say Phi2). I appreciate if you continue the above code.

Hesam

**Reply by ralfHielscher** [reply]
You see the plot command at the end of the script :)

**Comment by HPouraliakbar** [comment]
I got many errors for this

**Reply by ralfHielscher** [reply]
Most likely, you are not using the latest MTEX version.

**Comment by HPouraliakbar** [comment]
Can you please check the code below and let me know why I get errors:

ebsd = ebsd('indexed');
CS = crystalSymmetry('432')
sS = symmetrise(slipSystem.fcc(CS));
ori = ebsd.orientations;
epsilon = strainTensor(diag([-0.4 1 -0.6]))bsd = ebsd('indexed');
[M_ori,b_ori,W_ori] = calcTaylor(inv(ori)*epsilon,sS);
sP = phi2Sections(CS,specimenSymmetry('222'));
sP.phi2 = (0:5:90)*degree;
plot(M_ori,'smooth',sP)
mtexColorbar

This code calculates the M for all the indexed orientations but the plotting part does not work. I followed the following link:
https://mtex-toolbox.github.io/TaylorModel.html
to plot the fig.5 of mentioned paper
![TaylorModel_01](https://github.com/user-attachments/assets/f9fb965e-fa6a-4682-ae38-4063d6cb5299)

**Reply by ralfHielscher** [reply]
Your code is not correct. You can remove everything what is connected with `ebsd`. 

```matlab
CS = crystalSymmetry('432')

% define the slip system
sS = symmetrise(slipSystem.fcc(CS));

% define the strain
epsilon = strainTensor(diag([-0.4 1 -0.6]))

% define the plotting sections and the corresponding orientations
sP = phi2Sections(CS,specimenSymmetry('222'));
sP.phi2 = (0:5:90)*degree;
ori = sP.makeGrid;

% compute the Taylor factor
[M_ori,b_ori,W_ori] = calcTaylor(inv(ori)*epsilon,sS);

% do the plot
sP.plot(M_ori, 'smooth')
mtexColorbar
```

Please try to understand the code. It is not that hard.

Ralf

**Comment by HPouraliakbar** [comment]
Hi Ralph,

Thank you very much for your prompt attention.

I tried your code and unfortunately I got the following errors:

I tried two different MTEX versions (5.11.2 and 5.8.1) and both return these error lines. I think the "ori = sP.makeGrid;" line before calculating Taylor factor bring this. Can we calculate the Taylor factors for all orientations and then we use the makeGrid? My material is Aluminum with FCC structure.

Hesam

---------------------------
cs1 = orthorhombic specimenSymmetry  
cs2 = triclinic specimenSymmetry  
 Error using symmetry/ensureCS (line 55)

The above two symmetries does not match!

Error in tensor/rotate_outer (line 18)
  R = T.CS.ensureCS(R);

Error in  *  (line 40)
  r = rotate_outer(b,a);
---------------------------

**Comment by ralfHielscher** [comment]
Ok, lets give it a next try :)

```matlab
CS = crystalSymmetry('432')

% define the slip system
sS = symmetrise(slipSystem.fcc(CS));

% define the strain
epsilon = strainTensor(diag([-0.4 1 -0.6]))

% compute the Taylor factor
M = calcTaylor(epsilon,sS);

% define the plotting sections and the corresponding orientations
sP = phi2Sections(CS,specimenSymmetry('222'));
sP.phi2 = (0:15:90)*degree;

% do the plot
plot(M,sP)
mtexColorbar
```
Ralf.

**Comment by HPouraliakbar** [comment]
Hi Ralf,

Thank you so much for your kind help. The last code finally worked. 
Is it possible to plot Schmit Factor with the ODF sections as well?

Thanks in advance for your prompt attention.

Hesam
